### PR TITLE
added reduce mixin to performance report

### LINKED ISSUE #1490: [FEAT] Create the as_scalar mixin functionality

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1490

### CODE REVIEW

File: deepchecks/vision/utils/image_functions.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def prepare_thumbnail(
         image = ensure_image(image, copy=False)
 
     img_bytes = io.BytesIO()
-    image.save(fp=img_bytes, format='PNG')
+    image.save(img_bytes, optimize=True, quality=60, format='jpeg')
```
Comment (nirhutnik) [inline deepchecks/vision/utils/image_functions.py:192]: This is on purpose?

File: deepchecks/core/checks.py [hunk 2]
```diff
@@ -45,6 +45,14 @@ class CheckMetadata(TypedDict):
     summary: str
 
 
+class ReduceMixin(abc.ABC):
+    """Mixin for reduce_output function."""
+
+    def reduce_output(self, check_result: 'check_types.CheckResult') -> Dict[str, int]:
+        """Return the check result as a reduced dict."""
```
[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/core/checks.py:52]: I think the type hint says the general format, for each check the meaning of the str can be diffrent.

File: deepchecks/vision/utils/image_functions.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/vision/utils/image_functions.py:192]: this change should be in another pr probably

File: deepchecks/core/checks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ItayGabbay) [reply deepchecks/core/checks.py:52]: I think it's worth having a full docstring here, with Parameters and Returns sections.

[the remark this answers is not in the document]
Comment (JKL98ISR) [reply deepchecks/core/checks.py]: did docstring - very pretty

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**